### ISSUE TITLE
Clean up secp256k1-zkp dependencies

### ISSUE THREAD

**Issue by ignopeverell** [ignopeverell/grin#124]
Right now we have a full copy of the rust-secp256k1 source as well as a full copy of libsecp256k1-zkp source into a repository, with various modifications applied on top of both. It's messy, hard to track and doesn't allow us to merge from upstream. We need to clean that up.

Here's the proposed cleanup solution:

- Create 2 repos under github.com/mimblewimble, one for the C source and one for Rust.
- Create `upstream` vendor branches in both of them [1] that we keep in sync with the 2 upstream.
- Merge the `upstream` into master and apply our various changes on master.
- Have a standard crate dependency in grin (with a git link) to the new rust-secp256k1-zkp.

**Comment by antiochp** [comment]
Are these the upstream repos we want to be tracking?

* https://github.com/ElementsProject/secp256k1-zkp
* https://github.com/apoelstra/rust-secp256k1

**Comment by ignopeverell** [comment]
Correct.

**Comment by antiochp** [comment]
Quick update on where we are so far (putting it down here so I can refer back to this later...) -

I see we have a new repo created here - 
https://github.com/mimblewimble/secp256k1-zkp

I forked this to my own github here - 
https://github.com/antiochp/secp256k1-zkp

Right now I have 2 branches on my own fork - `master` and `upstream`.
And 2 upstream remotes setup - 

```
git remote -v
origin	https://github.com/antiochp/secp256k1-zkp.git (fetch)
origin	https://github.com/antiochp/secp256k1-zkp.git (push)
upstream_elements	https://github.com/ElementsProject/secp256k1-zkp.git (fetch)
upstream_elements	https://github.com/ElementsProject/secp256k1-zkp.git (push)
upstream_mimblewimble	https://github.com/mimblewimble/secp256k1-zkp.git (fetch)
upstream_mimblewimble	https://github.com/mimblewimble/secp256k1-zkp.git (push)
```

I merged the `secp256k1-zkp` branch from `ElementsProject` onto my `upstream` branch.
Then merged this onto my `master` branch to give us a baseline.

Currently working on porting across the changes that we have made to the copy of this repo that we have in grin onto this `master` branch.
There are 2 changes - 
https://github.com/ignopeverell/grin/commit/ca496e (relatively minor change)
https://github.com/ignopeverell/grin/commit/b5e62c (bigger change)

@ignopeverell what's the best way of getting this onto the mimblewimble/secp256k1-zkp repo once this is all ready to go?
You want me to open a PR(s) for this?

**Comment by antiochp** [comment]
One outstanding question - 
once we do something similar for the `rust-secp256k1` repo, I'm not sure how to best approach the dependency *between* these 2 repos.
We may just want to take `secp256k1-zkp` and "vendor" it in to `rust-secp256k1` (and check it into the rust git repo). Then just keep this updated manually (like it is done currently in the various copies of rust-secp256k1).

I'm not sure if there is a better approach to this?

**Comment by ignopeverell** [comment]
Ooops, I've just created the vendor branch and merged to master on the new secp256k1-zkp. Ideally yes, you'd do a PR against that but I'm not sure whether you can do it now from your fork...

My plan was to use a git submodule to our secp256k1-zkp in our rust-secp256k1-zkp.

**Comment by ignopeverell** [comment]
I've created rust-secp256k1-zkp as well, removing the hard check-in of secp256k1 and replacing it by a submodule to secp256k1-zkp.

**Comment by antiochp** [comment]
Update on where we are with this - 

PR mimblewimble/secp256k1-zkp#5 that ports across `secp256k1_pedersen_commit_sum` (required by Grin).

PR mimblewimble/rust-secp256k1-zkp#2 that ports over all the functions required by Grin and reworks the code to work with the more recent `secp256k1-zkp` code.

PR ignopeverell/grin/pull/126 that removes checked in secp256k1zkp and specifies it as external dependency in Cargo.toml

**Comment by antiochp** [comment]
Resolved by ignopeverell/grin#126
